### PR TITLE
Clean up trigger rules and branch filters

### PR DESCRIPTION
**Pull request by osipxd**
- Fixed the format of branch filters - full reference where needed and a logical name for the rest
- Enabled approvals for untrusted builds for the sake of security (be ready to approve contributors' builds)
- Applied the pull-requests feature to all builds that should be triggered by pull requests, so now it should be possible to use `+pr:*` syntax in rules
- Gathered trigger rules and branch filters into objects to make them simpler to reuse
- Disabled ignoring for bot commits
- Aligned trigger branch rules of builds with the composite build to prevent builds on branches without PRs

It is simpler to review commit-by-commit

I hope after these changes we will get back running composite build on the main branch and get rid of builds running on branches without PRs.

### CODE REVIEW

File: .teamcity/settings.kts [hunk 1]
```diff
@@ -60,6 +61,24 @@ project {
...
+            manualRunsApproved = true
+            enableLog = true
+        }
+
+        youtrack {
+            id = "YouTrack"
+            displayName = "JetBrains YouTrack"
+            host = "https://youtrack.jetbrains.com/"
+            projectExtIds = "KTOR"
+            accessToken = "%youtrack.token%"
```
Comment (osipxd) [inline .teamcity/settings.kts:78]: Is it possible to generate a non-personal token for YouTrack?

Comment (e5l) [reply .teamcity/settings.kts:78]: nope

**Review by e5l — APPROVED** [review]
thanks for cleaning it up. now it looks better than the original teamcity magic

File: .teamcity/settings.kts (hunk 1 above)
Comment (osipxd) [reply .teamcity/settings.kts:78]: How can I add a property to CI?

**Pull request merged**